{{ cookiecutter.format }}/{{ cookiecutter.app_name }}.wxs

```diff
@@ -38,141 +23,293 @@
+                    <RegistryValue
+                        Root="HKMU"
+                        Key="Software\{{ cookiecutter.author or 'Unknown Developer' }}\{{ cookiecutter.formal_name }}"
+                        Name="installed"
+                        Type="integer"
+                        Value="1"
+                        KeyPath="yes" />
+                    <RemoveFolder Id="ProgramMenuSubfolder" On="uninstall" />
+                </Component>
             </Directory>
-        </Directory>
+        </StandardDirectory>
 
         {% if cookiecutter.document_types -%}
-        <SetProperty Id='FileAssociationProperty' Value="[{{ cookiecutter.module_name }}_ROOTDIR]{{ cookiecutter.formal_name }}.exe" After="CostFinalize"/>
+        <SetProperty
+            Id="FileAssociationProperty"
+            Value="[APPLICATIONFOLDER]{{ cookiecutter.binary_path }}"
+            After="CostFinalize" />
 
         {% for document_type_id, document_type in cookiecutter.document_types|dictsort -%}
-        <Component Id="FileAssociation.{{ document_type_id }}" Directory="{{ cookiecutter.module_name }}_ROOTDIR" Guid="*">
-            <File Id="ProductIcon.{{ document_type_id }}" Source="{{ cookiecutter.app_name }}-{{ document_type_id }}.ico" />
-            <ProgId Id="MyApp.{{ document_type_id }}" Description="{{ document_type.description }}" Icon="ProductIcon.{{ document_type_id }}">
-                <Extension Id="{{ document_type.extension }}" ContentType="{% if document_type.get('mime_type') %}{{ document_type.mime_type }}{% else %}application/x-{{ cookiecutter.app_name }}-{{ document_type_id }}{% endif %}">
-                    <Verb Id="open" Command="Open" TargetProperty="FileAssociationProperty" Argument='"%1"' />
+        <Component
+            Id="FileAssociation.{{ document_type_id }}"
+            Directory="APPLICATIONFOLDER">
+            <File
+                Id="ProductIcon.{{ document_type_id }}"
+                Source="{{ cookiecutter.app_name }}-{{ document_type_id }}.ico" />
+            <ProgId
+                Id="MyApp.{{ document_type_id }}"
+                Description="{{ document_type.description }}"
+                Icon="ProductIcon.{{ document_type_id }}">
+                <Extension
+                    Id="{{ document_type.extension }}"
+                    ContentType="{% if document_type.get('mime_type') %}{{ document_type.mime_type }}{% else %}application/x-{{ cookiecutter.app_name }}-{{ document_type_id }}{% endif %}">
+                    <Verb
+                        Id="open"
+                        Command="Open"
+                        TargetProperty="FileAssociationProperty"
+                        Argument='"%1"' />
                 </Extension>
             </ProgId>
         </Component>
         {%- endfor %}
         {%- endif %}
 
-        <InstallExecuteSequence>
-            <RemoveExistingProducts After="InstallValidate"/>
-        </InstallExecuteSequence>
```

Comment:
The `RemoveExistingProducts` element is no longer required because WiX now [handles this by default](https://docs.firegiant.com/wix/whatsnew/#default-major-upgrades).